### PR TITLE
Remove unnecessary checks

### PR DESCRIPTION
**Pull request by egorovdanil**
[https://github.com/Intel-Media-SDK/MediaSDK/issues/3](https://github.com/Intel-Media-SDK/MediaSDK/issues/3)
[https://github.com/Intel-Media-SDK/MediaSDK/issues/5](https://github.com/Intel-Media-SDK/MediaSDK/issues/5)

### CODE REVIEW

File: _studio/mfx_lib/plugin/src/mfx_hevc_enc_plugin.cpp [hunk 1]
```diff
@@ -93,10 +93,9 @@ mfxStatus MFXHEVCEncoderPlugin::PluginInit(mfxCoreInterface *core)
 
 mfxStatus MFXHEVCEncoderPlugin::PluginClose()
 {
-    if (m_encoder) {
-        delete m_encoder;
-        m_encoder = NULL;
-    }
+    delete m_encoder;
+    m_encoder = NULL;
```
Comment (mgonchar) [inline _studio/mfx_lib/plugin/src/mfx_hevc_enc_plugin.cpp]: use ```nullptr```

File: samples/sample_common/include/avc_headers.h [hunk 2]
```diff
@@ -49,11 +49,8 @@ class HeaderSet
             m_header.resize(id + 1);
         }
 
-        if (m_header[id])
-        {
-            delete m_header[id];
-            m_header[id]=0;
-        }
+        delete m_header[id];
+        m_header[id]=0;
```
Comment (mgonchar) [inline samples/sample_common/include/avc_headers.h]: you are assigning new value in next line. No need to zero it anyway (especially with ```int``` literal)

File: _studio/shared/umc/codec/jpeg_common/src/bitstreamout.cpp [hunk 3]
```diff
@@ -57,11 +57,8 @@ JERRCODE CBitStreamOutput::Attach(CBaseStreamOutput* out)
 
 JERRCODE CBitStreamOutput::Detach(void)
 {
-  if(m_pData)
-  {
-    // deallocate internal memory
-    delete[] m_pData;
-  }
+  // deallocate internal memory
```
Comment (mgonchar) [inline _studio/shared/umc/codec/jpeg_common/src/bitstreamout.cpp]: this is really useless comment :)

File: _studio/mfx_lib/plugin/src/mfx_hevc_enc_plugin.cpp [hunk 4]
```diff
@@ -1,4 +1,4 @@
-// Copyright (c) 2017 Intel Corporation
+// Copyright (c) 2019 Intel Corporation
```
Comment (mgonchar) [inline _studio/mfx_lib/plugin/src/mfx_hevc_enc_plugin.cpp]: often we do ```2017-2019``` in such cases. But I don't think if particular format matters. @onabiull can you comment here?

**Comment by egorovdanil** [comment]
> @egorovdanil : please, add reference to the addressed issues in **commit message**, not in PR.

Done

File: samples/sample_common/include/avc_headers.h [hunk 5]
```diff
@@ -49,11 +49,7 @@ class HeaderSet
             m_header.resize(id + 1);
         }
 
-        if (m_header[id])
-        {
-            delete m_header[id];
-            m_header[id]=0;
-        }
+        delete m_header[id];
```
Comment (dvrogozh) [inline samples/sample_common/include/avc_headers.h]: @mgonchar : why we don't manage this memory via std::unique_ptr? Few other places in the implementation of this stuff raises questions, for example RefCounter class here: https://github.com/Intel-Media-SDK/MediaSDK/blob/master/samples/sample_common/include/avc_structures.h#L223

File: _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp [hunk 6]
```diff
@@ -5009,7 +5009,7 @@ mfxStatus MfxHwH264Encode::CheckMVCSeqDescQueryLike(mfxExtMVCSeqDesc * mvcSeqDes
         mvcSeqDesc->NumView = 0;
     }
 
-    if (mvcSeqDesc->NumOP > 0 && mvcSeqDesc->NumOP > 1024)
+    if (mvcSeqDesc->NumOP > 1024)
```
Comment (dvrogozh) [inline _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp]: This couple of lines above:
`if (mvcSeqDesc->NumView > 0 && (mvcSeqDesc->NumView > 2 || mvcSeqDesc->NumView < 2))`

Is it the same as:
`if (mvcSeqDesc->NumView > 0 && mvcSeqDesc->NumView != 2)`

Or I don't understand something?

Comment (egorovdanil) [reply _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp]: done

File: samples/sample_common/include/avc_headers.h (hunk 5 above)
Comment (mgonchar) [reply samples/sample_common/include/avc_headers.h]: @dvrogozh ```unique_ptr``` is always better for memory management) but it should be subject of another PR.

yes, code you referred to is looking as self invented ```std::shared_ptr```

**Comment by mgonchar** [comment]
@dvrogozh I've created [issue ](https://github.com/Intel-Media-SDK/MediaSDK/issues/1739) about self-invented shared ptr, which you've mentioned in your comment

File: samples/sample_common/include/avc_headers.h (hunk 5 above)
Comment (dvrogozh) [reply samples/sample_common/include/avc_headers.h]: Let's track and address this in #1739.

**Pull request merged**